Updates to Java docs per comments in Hotjar (#414)
temporalio/documentation: docs/java/workflows.md

- **@WorkflowMethod** denotes the starting point of Workflow execution. Workflow execution completes when this methods returns.
- **@SignalMethod** indicates that this method is a signal handler method and that it can react to external signals. It can have parameters which can contain the signal payload. It does not return a value, so it must have a `void` return type.

> - **@QueryMethod** indicates that this method can be used to query the workflow state any time during its execution. It can have parameters which can be used to filter a subset of the workflow state it returns. Since it does return a value it must have a non `void` return type.

Reviewer: ```suggestion
- **@QueryMethod** indicates that this method can be used to query the Workflow's state at any time during its execution.
It can have parameters which can be used to filter a subset of the Workflow's state that it returns. Since it does return a value it must have a non `void` return type.
```

The `@WorkflowMethod` annotation has a `name` parameter, for example: `@WorkflowMethod(name = "MyWorkflowType")`.
It can be used to denote the Workflow type. If not set, the Workflow type defaults to the short name of the Workflow interface,
in the example above being `FileProcessingWorkflow`.
Methods annotated with `@WorkflowMethod` can have any number of parameters.

> We recommend passing a single parameter that contains all the input fields. This allows adding fields in a backward compatible manner.

Reviewer: ```suggestion
We recommend passing a single parameter that contains all the input fields.
This allows adding fields in a backward compatible manner.
```

(depending on which Workflow type has been started) annotated with `@WorkflowMethod` can be invoked.
As soon as this method returns, the Workflow execution is considered as completed.

> Workflow methods annotated with `@QueryMethod` and `@SignalMethod` can be invoked during Workflow execution.

Reviewer: ```suggestion
Workflow methods annotated with `@QueryMethod` and `@SignalMethod` can be invoked during a Workflow's execution.
```

Note that methods annotated with `@QueryMethod` can be invoked even when a Workflow is in the `Completed` 

> state.

Reviewer: I want to see each sentence on its own line, makes it way easier to review changes.
Does your editor have a max width that causes these to wrap ?

Author: probably, i will fix it. sorry
